### DISCUSSION TITLE
module 'jsbsim' has no attribute 'FGRocket'

### DISCUSSION THREAD

**Discussion by IMHinnG** [JSBSim-Team/jsbsim#696] [Support requests]
attribute 'FGRocket' non-existent in python

Traceback (most recent call last):
  File "JSBSim_1.py", line 6, in <module>
    jsbsim.FGRocket('..\\')
AttributeError: module 'jsbsim' has no attribute 'FGRocket'

**Comment by seanmcleod** [comment]
What are you trying to do? Share the contents of `JSBSim_1.py` or least the relevant sections that are trying to use `FGRocket`.

**Comment by IMHinnG** [comment]
> What are you trying to do? Share the contents of `JSBSim_1.py` or least the relevant sections that are trying to use `FGRocket`.

its appears in c++ code, but not in python  of this attribute,

**Comment by seanmcleod** [comment]
Not every single C++ class and method is exposed 1 for 1 in the python API.

Hence my question:

> What are you trying to do? Share the contents of `JSBSim_1.py` or least the relevant sections that are trying to use `FGRocket`.

If you share that sort of information we can potentially help in terms of possibly pointing you to an alternative API to achieve the same goal, or if there is a good reason for exposing something via the python API then we can look at that as well.

But we need more information.

**Comment by IMHinnG** [comment]
> Not every single C++ class and method is exposed 1 for 1 in the python API.
> 
> Hence my question:
> 
> > What are you trying to do? Share the contents of `JSBSim_1.py` or least the relevant sections that are trying to use `FGRocket`.
> 
> If you share that sort of information we can potentially help in terms of possibly pointing you to an alternative API to achieve the same goal, or if there is a good reason for exposing something via the python API then we can look at that as well.
> 
> But we need more information.

```python

import jsbsim

jsbsim.FGRocket('..\')

```

that is all

**Reply by bcoconni** [reply]
> ```python
> import jsbsim
> 
> jsbsim.FGRocket('..\')
> ```
> 
> that is all

Even in C++, `FGRocket` is not meant to be used in isolation and there is no constructor of `FGRocket` that takes a string as an input so don't expect this code to work at all.

The simplest example of using a rocket with JSBSim is:
```python
import jsbsim

fdm = jsbsim.FGFDMExec(None)  # Use JSBSim default aircraft data.
fdm.load_script('scripts/J2461.xml')
fdm.run_ic()

while fdm.run():
  pass
```

**Reply by IMHinnG** [reply]
thanks

**Comment by seanmcleod** [comment]
That's not enough information for me to make any useful comments or suggestions.
